bins/fuzzed/clusterfuzz-testcase-minimized-ia_fuzz-6301506113634304 - consistent TIMEOUT

See https://github.com/radareorg/radare2/runs/840098234#step:13:1095
```
make[1]: Leaving directory '/home/runner/work/radare2/radare2/test/unit'
r2r -LF bins/fuzzed @fuzz
Loaded 505 tests.


[XX] TIMEOUT /home/runner/work/radare2/radare2/test/bins/fuzzed <fuzz> /home/runner/work/radare2/radare2/test/bins/fuzzed/clusterfuzz-testcase-minimized-ia_fuzz-6301506113634304
R2_NOPLUGINS=1 radare2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Qc aaa /home/runner/work/radare2/radare2/test/bins/fuzzed/clusterfuzz-testcase-minimized-ia_fuzz-6301506113634304
-- stdout

-- stderr
Warning: truncated typeidx buffer
This DEX is probably corrupted. Chopping DM from 1362815431 to 4KB

-- exit status: -1
[**] /home/runner/work/radare2/radare2/test/bins/fuzzed      504 OK         0 BR        1 XX        0 FX

Finished in 24 minutes and 32 seconds.
make: *** [fuzz-tests] Error 1
Makefile:15: recipe for target 'fuzz-tests' failed
##[error]Process completed with exit code 2.
```

--- a/libr/bin/p/bin_dex.c
+++ b/libr/bin/p/bin_dex.c
@@ -1150,10 +1150,6 @@ static void parse_dex_class_method(RBinFile *bf, RBinDexClass *c, RBinClass *cls
 	if (!dex->trycatch_list) {
 		dex->trycatch_list = r_list_newf ((RListFree)r_bin_trycatch_free);
 	}
-	if (DM > 4096) {
-		eprintf ("This DEX is probably corrupted. Chopping DM from %d to 4KB\n", (int)DM);
-		DM = 4096;
-	}
 	size_t skip = 0;
 	ut64 bufsz = r_buf_size (bf->buf);
 	ut64 encoded_method_addr;
@@ -1165,9 +1161,17 @@ static void parse_dex_class_method(RBinFile *bf, RBinDexClass *c, RBinClass *cls
 		// Needed because theres another rbufseek call inside this loop. must be fixed
 		encoded_method_addr = r_buf_tell (bf->buf);
 		MI = peek_uleb (bf->buf, &err, &skip);
+		if (err) {
+			eprintf ("Error\n");
+			break;
+		}
 		MI += omi;
 		omi = MI;
 		MA = peek_uleb (bf->buf, &err, &skip);
+		if (err) {
+			eprintf ("Error\n");
+			break;
+		}
 		MC = peek_uleb (bf->buf, &err, &skip);
 		if (err) {
 			eprintf ("Error\n");
@@ -1208,9 +1212,24 @@ static void parse_dex_class_method(RBinFile *bf, RBinDexClass *c, RBinClass *cls
 				continue;
 			}
 			regsz = r_buf_read_le16_at (bf->buf, MC);
+			if (regsz == UT16_MAX) {
+				R_FREE (flag_name);
+				R_FREE (signature);
+				break;
+			}
 			ins_size = r_buf_read_le16_at (bf->buf, MC + 2);
+			if (ins_size == UT16_MAX) {
+				R_FREE (flag_name);
+				R_FREE (signature);
+				break;
+			}
 			outs_size = r_buf_read_le16_at (bf->buf, MC + 4);
 			tries_size = r_buf_read_le16_at (bf->buf, MC + 6);
+			if (tries_size == UT16_MAX) {
+				R_FREE (flag_name);
+				R_FREE (signature);
+				break;
+			}
 			debug_info_off = r_buf_read_le32_at (bf->buf, MC + 8);
 			insns_size = r_buf_read_le32_at (bf->buf, MC + 12);
 			int padd = 0;
@@ -1284,7 +1303,9 @@ static void parse_dex_class_method(RBinFile *bf, RBinDexClass *c, RBinClass *cls
 					}
 					// TODO: catch left instead of null
 					st64 size;
-					r_buf_seek (bf->buf, off, R_BUF_SET);
+					if (r_buf_seek (bf->buf, off, R_BUF_SET) == -1) {
+						break;
+					}
 					int r = r_buf_sleb128 (bf->buf, &size);
 					if (r <= 0) {
 						break;

